map/framework.cpp

```diff
@@ -2472,7 +2474,7 @@ void Framework::SetRouterImpl(RouterType type)
 
     router.reset(
         new CarRouter(m_model.GetIndex(), countryFileGetter,
-                      SingleMwmRouter::CreateCarRouter(m_model.GetIndex())));
+                      SingleMwmRouter::CreateCarRouter(m_model.GetIndex(), m_routingSession)));
```

Comment:
There is some kind of circular dependency now, right? Because router is then passed to m_routingSession.